pkg/auth/providers/azure/azure_provider_test.go

```diff
@@ -225,3 +228,84 @@ func TestTransformToAuthProvider(t *testing.T) {
 		})
 	}
 }
+
+func TestMigrateNewFlowAnnotation(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name               string
+		current            *v3.AzureADConfig
+		proposed           *v3.AzureADConfig
+		annotationExpected bool
+	}{
+		{
+			name: "new setup on Rancher v2.6.7+ after an upgrade from previous version",
+			current: &v3.AzureADConfig{
+				AuthConfig: v3.AuthConfig{
+					Enabled: false,
+				},
+				GraphEndpoint: "https://graph.microsoft.com",
+			},
+			proposed:           &v3.AzureADConfig{},
+			annotationExpected: true,
+		},
+		{
+			name: "new setup on Rancher v2.6.7+",
+			current: &v3.AzureADConfig{
+				AuthConfig: v3.AuthConfig{
+					Enabled: false,
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{
+							GraphEndpointMigratedAnnotation: "true",
+						},
+					},
+				},
+				GraphEndpoint: "https://graph.microsoft.com",
+			},
+			proposed:           &v3.AzureADConfig{},
+			annotationExpected: true,
+		},
+		{
+			name: "reconfigure existing deprecated setup",
+			current: &v3.AzureADConfig{
+				AuthConfig: v3.AuthConfig{
+					Enabled: true,
+				},
+				GraphEndpoint: "https://graph.windows.net/",
+			},
+			proposed:           &v3.AzureADConfig{},
+			annotationExpected: false,
+		},
+		{
+			name: "reconfigure existing new setup",
+			current: &v3.AzureADConfig{
+				AuthConfig: v3.AuthConfig{
+					Enabled: true,
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{
+							GraphEndpointMigratedAnnotation: "true",
+						},
+					},
+				},
+				GraphEndpoint: "https://graph.microsoft.com",
+			},
+			proposed:           &v3.AzureADConfig{},
+			annotationExpected: true,
+		},
+	}
+
+	for _, test := range tests {
```

Comment:
If we loop by value and call `Parallel()` test variable will be overridden.
```suggestion
	for i := range tests {
	    test := tests[i]
```